src/persist/src/unreliable.rs

```diff
@@ -106,9 +106,10 @@ impl<U: Buffer> Buffer for UnreliableBuffer<U> {
         self.buf.truncate(upper)
     }
 
-    fn close(&mut self) -> Result<(), Error> {
+    fn close(&mut self) -> Result<bool, Error> {
```

Comment:
this seems like it flips the semantics from before? previously we checked availability before attempting to close, now we always attempt to close but only report success if available. not sure what the rationale for that was and it might be worth a comment?